### ISSUE TITLE
Nested mixins don't work

### ISSUE THREAD

**Issue by SCIF** [vimeo/psalm#5719]
https://psalm.dev/r/f31ffcdffb

Not sure if that's intended to be working but since there is no documentation about it, I want to ask that question.

**Comment by AlbinoDrought** [comment]
Similarly: https://psalm.dev/r/1b9a1fb99a

(experiencing this when using https://github.com/laravel/framework/blob/9.x/src/Illuminate/Contracts/Database/Eloquent/Builder.php#L10 )

**Comment by alies-dev** [comment]
Still reproduces on Psalm `7.0.0-beta19`. The root issue is that `@mixin` is only resolved one level deep: a method that lives on a transitively mixed-in class (depth ≥ 2) is reported as undefined.

Minimal repro:

```php
<?php
namespace T;

class QueryB { public function whereNull(string $c): static { return $this; } }

/** @mixin \T\QueryB */
class EloquentB {}

/** @mixin \T\EloquentB */
interface IfaceContract {}

/** @mixin \T\EloquentB */
class ClassHost {}

function direct(EloquentB $q): void   { $q->whereNull('x'); } // OK   (depth 1)
function viaClass(ClassHost $q): void { $q->whereNull('x'); } // ERROR (depth 2)
function viaIface(IfaceContract $q): void { $q->whereNull('x'); } // ERROR (depth 2)
```

Output:

```
UndefinedMethod:          Method T\ClassHost::whereNull does not exist
UndefinedInterfaceMethod: Method T\IfaceContract::whereNull does not exist
```

So the bug is not interface-specific. Depth-1 mixin resolves; depth-2 does not, for both class and interface hosts.

### Why this matters in practice

Laravel 12 ships query builder contracts that are intentionally empty and rely entirely on a transitive `@mixin` chain:

```php
// Illuminate\Contracts\Database\Eloquent\Builder
/** @mixin \Illuminate\Database\Eloquent\Builder */
interface Builder extends \Illuminate\Contracts\Database\Query\Builder {}

// Illuminate\Database\Eloquent\Builder
/** @mixin \Illuminate\Database\Query\Builder */   // where whereNull(), whereIn(), orderBy(), withTrashed()… live
class Builder { /* ... */ }
```

Any code type-hinting the contract (very common — e.g. Nova filters/lenses receive `Illuminate\Contracts\Database\Eloquent\Builder`) gets `UndefinedInterfaceMethod` for every base query-builder method, because resolution stops at the first `@mixin` hop. The same empty-contract-plus-`@mixin` pattern is used by other Laravel contracts (Pagination, Session, View, Cache), so a single fix here would clear a large class of false positives for Laravel projects.

### Prior art

Two earlier attempts at deep/recursive mixin resolution were opened and closed unmerged: #11082 ("Resolving deep mixins") and #6571 ("WIP: Mixin magic methods").
